packages/volto/src/components/manage/Preferences/ChangePassword.jsx

```diff
@@ -70,169 +61,97 @@ const messages = defineMessages({
   },
 });
 
-/**
- * ChangePassword class.
- * @class ChangePassword
- * @extends Component
- */
-class ChangePassword extends Component {
-  /**
-   * Property types.
-   * @property {Object} propTypes Property types.
-   * @static
-   */
-  static propTypes = {
-    userId: PropTypes.string.isRequired,
-    loading: PropTypes.bool.isRequired,
-    updatePassword: PropTypes.func.isRequired,
-    pathname: PropTypes.string.isRequired,
-  };
-
-  /**
-   * Constructor
-   * @method constructor
-   * @param {Object} props Component properties
-   * @constructs ChangePassword
-   */
-  constructor(props) {
-    super(props);
-    this.onCancel = this.onCancel.bind(this);
-    this.onSubmit = this.onSubmit.bind(this);
-    this.state = { isClient: false };
-  }
+const ChangePassword = () => {
+  const intl = useIntl();
+  const dispatch = useDispatch();
+  const isClient = useClient();
 
-  /**
-   * Component did mount
-   * @method componentDidMount
-   * @returns {undefined}
-   */
-  componentDidMount() {
-    this.setState({ isClient: true });
-  }
+  const userId = useSelector(
+    (state) =>
+      state.userSession.token ? jwtDecode(state.userSession.token).sub : '',
+    shallowEqual,
+  );
+  const loading = useSelector((state) => state.users.update_password.loading);
+  const { pathname } = useLocation();
+  const history = useHistory();
 
-  /**
-   * Submit handler
-   * @method onSubmit
-   * @param {object} data Form data.
-   * @returns {undefined}
-   */
-  onSubmit(data) {
+  const onSubmit = (data) => {
     if (data.newPassword === data.newPasswordRepeat) {
-      this.props.updatePassword(
-        this.props.userId,
-        data.oldPassword,
-        data.newPassword,
-      );
+      dispatch(updatePassword(userId, data.oldPassword, data.newPassword));
       toast.success(
```

Comment:
This should be dispatch(updatePassword( ... )).then(() => {
    toast.success( ... );
}

(We only want to show the toast message after the API call was completed successfully.)